common/page.go

```diff
@@ -421,6 +426,22 @@ func (e *MetricEvent) Tag(matchesRegex K6BrowserCheckRegEx, matches TagMatches)
 	}
 
 	for _, m := range matches.Matches {
+		// Validate the request method type if it has been assigned in a Match.
+		method := strings.TrimSpace(m.Method)
+		if method != "" {
+			method = strings.ToUpper(method)
+			switch method {
+			case http.MethodGet, http.MethodPost, http.MethodPut, http.MethodDelete, http.MethodPatch,
+				http.MethodHead, http.MethodOptions, http.MethodConnect, http.MethodTrace:
+			default:
+				return fmt.Errorf("method %q is invalid", m.Method)
+			}
```

Comment:
I'd move this to a helper function in helpers.go. I believe it's a bit out of place here and obscures the `Tag` method's actual logic flow.

Reply:
Thanks for the suggestion. I was wondering if this needed to be extracted out into a separate validation function. I think for now though i'm going to leave it as is.